pylds/base.py

```diff
@@ -6,32 +6,122 @@
     An additional dimension initiallised with zeros is added for the calculation of Lagrangian Descriptors.
-    
+
     Parameters
     ----------
-    grid_parameters : list of 3-tuples of floats
+    slice_parameters : list of 3-tuples of floats
         input parameters of limits and size of mesh per axis
-        
+
     Returns
     -------
     mesh : 1d numpy array
         flattened array of initial conditions
     """
-    x_min, x_max, Nx = grid_parameters[0]
-    y_min, y_max, Ny = grid_parameters[1]
-    points_x = np.linspace(x_min, x_max, Nx)
-    points_y = np.linspace(y_min, y_max, Ny)    
-    Y, X = np.meshgrid(points_y, points_x)  # Grid in phase-space
-    mesh = np.transpose([X.flatten(), Y.flatten(), np.zeros(Nx*Ny)]) # 2D grid + a zero column for LDs
-    return mesh.flatten()
+    if type(grid_parameters) == dict:
+        # Unpack extra grid parameters
+        slice_parameters = grid_parameters['slice_parameters']
+        dof_slice = grid_parameters['dof_slice']
+        dof_fixed, dof_fixed_values = grid_parameters['dof_fixed']
+        momentum_sign = grid_parameters['momentum_sign']
+        potential_energy = grid_parameters['potential_energy']
+        H0 = grid_parameters['energy_level']
+
+        N_dim = len(dof_slice)  # Total number of DoF
+        # Check N DoF is even.
+        if N_dim % 2 != 0:
+            error_mssg = ("ERROR: Number of DoF not even. ",
+                          "Check your extra grid parameters")
+            print(error_mssg)
+            sys.exit()
+        # Determine number of DoF for Energy conservation.
+        # There must be only one.
+        dof_remaining = 1 - (np.array(dof_fixed) + np.array(dof_slice))
+        if list(dof_remaining).count(1) > 1:
+            error_mssg = ("ERROR: More than one remaing DoF. ",
+                          "Cannot use Energy conservation to define high-dim grid.")
+            print(error_mssg)
+            sys.exit()
+
+        # Axes of visualisation slice
+        def pass_axis_parameters(parameters): return np.linspace(*parameters)
+        points_slice_axes = list(map(pass_axis_parameters, slice_parameters))
+        slice_mesh = np.meshgrid(*points_slice_axes)
+        dof_slice_axes = [axis.flatten() for axis in slice_mesh]
+
+        # Axes of fixed DoF
+        dof_fixed_axes = dof_fixed_values
+
+        N_points_slice = np.prod(list(map(itemgetter(-1), slice_parameters)))
+        phase_space_axes = {}
+
+        # Define and sort axes in phase space
+        k = 0
+        for i in range(N_dim):
+            if dof_slice[i] == 1:
+                phase_space_axes[i] = dof_slice_axes[k]
+                k += 1
+        k = 0
+        for i in range(N_dim):
+            if dof_fixed[i] == 1:
+                phase_space_axes[i] = dof_fixed_axes[k] * \
+                    np.zeros(N_points_slice)
+                k += 1
```

Comment:
Some of your later commits must have changed `np.ones` back to `np.zeros`. Can you verify that other fixes survived?

Reply:
I've fixed the zeros to ones again, but other updates need verifying

Reply:
I see. Let me check this too because I'm getting merging conflicts. 
Btw, seems like the EXAMPLES notebook that you modified was an outdated version.